connector/src/main/scala/com/datastax/spark/connector/datasource/CassandraCatalog.scala

```diff
@@ -0,0 +1,431 @@
+          createStmt.withSimpleStrategy(replicationFactor.toInt)
+        case NetworkTopologyStrategy =>
+          val datacenters = (ksMeta -- IgnoredReplicationOptions).map(pair => (pair._1, pair._2.toInt: java.lang.Integer))
+          createStmt.withNetworkTopologyStrategy(datacenters.asJava)
+        case other => throw new CassandraCatalogException(s"Unknown keyspace replication strategy $other")
+      }
+      val finalCreateStmt = createWithReplication.withDurableWrites(ksMeta.getOrElse(DurableWrites, "True").toBoolean)
+      session.execute(finalCreateStmt.asCql())
+    }
+  }
+
+  override def alterNamespace(namespace: Array[String], changes: NamespaceChange*): Unit = {
+    checkNamespace(namespace)
+
+    val ksMeta: mutable.Map[String, String] = changes.foldRight(loadNamespaceMetadata(namespace).asScala) {
+      case (setProperty: SetProperty, metadata: mutable.Map[String, String]) =>
+        metadata + (setProperty.property() -> setProperty.value)
+      case (removeProperty: RemoveProperty, metadata: mutable.Map[String, String]) =>
+        metadata - removeProperty.property()
+      case (other, _) => throw new CassandraCatalogException(s"Unable to handle alter namespace operation: ${other.getClass.getSimpleName}")
+    }
+
+    val alterStart = SchemaBuilder.alterKeyspace(namespace.head)
+    val alterWithDurable = alterStart.withDurableWrites(ksMeta.getOrElse(DurableWrites, "True").toBoolean)
+    val replicationClass = ksMeta
+      .getOrElse(ReplicationClass, throw new CassandraCatalogException(s"Creating a keyspace requires a $ReplicationClass option"))
+      .split("\\.")
+      .last
+    val alterWithReplication = replicationClass.toLowerCase(Locale.ROOT) match {
+      case SimpleStrategy =>
+        val replicationFactor = ksMeta.getOrElse(ReplicationFactor,
+          throw new CassandraCatalogException(s"Need a $ReplicationFactor option with SimpleStrategy"))
+        alterWithDurable.withSimpleStrategy(replicationFactor.toInt)
+      case NetworkTopologyStrategy =>
+        val datacenters = (ksMeta -- IgnoredReplicationOptions).map(pair => (pair._1, pair._2.toInt: java.lang.Integer))
+        alterWithDurable.withNetworkTopologyStrategy(datacenters.asJava)
+      case other => throw new CassandraCatalogException(s"Unknown replication strategy $other")
+    }
+
+    connector.withSessionDo(session =>
+      session.execute(alterWithReplication.asCql())
+    )
+  }
+
+  override def loadNamespaceMetadata(namespace: Array[String]): java.util.Map[String, String] = {
+    val ksMetadata = getKeyspaceMeta(connector, namespace)
+
+    (ksMetadata.getReplication.asScala + (DurableWrites -> ksMetadata.isDurableWrites.toString))
+      .asJava
+  }
+
+  override def dropNamespace(namespace: Array[String]): Boolean = {
+    checkNamespace(namespace)
+    val keyspace = getMetadata(connector).getKeyspace(fromInternal(namespace.head)).asScala
```

Comment:
FromInternal takes the unquoted mixed case version of the name and quotes it if necessary. So I believe this will NOT match an improperly cased table in SparkSql but I think if spark sql naming is case sensitive we should respect that ... ?